### ISSUE TITLE
[Bug] Subgraph doesn't sync after node restart

### ISSUE THREAD

**Issue by mehditvk8** [graphprotocol/graph-node#4921]
### Bug report

I ran graph-node locally on my server and I deployed Ethereum subgraph on that. It works and syncs well. But when I stop graph-node and run it again, the subgraph stops syncing until deploy it again. I tried it with different RPCs and chains again. But I saw the same problem on them. Anyone have any idea  why does this happen? I it normal or graph-node bug?

### Some information to help us out

- [ ] Tick this box if this bug is caused by a regression found in the latest release.
- [ ] Tick this box if this bug is specific to the hosted service.
- [X] I have searched the issue tracker to make sure this issue is not a duplicate.

### OS information

Linux

**Comment by azf20** [comment]
Hi @mehditvk8 that doesn't sound normal, can you share any logs from the Graph Node? What version are you running?

**Comment by mehditvk8** [comment]
@azf20 As I see while inspecting the docker image, It is the latest version.

>         "Id": "sha256:8aa859b2b214d8059d09f971c2b808b76e2fe854946511186e8780c8be884a43",
>        "RepoTags": [
>            "graphprotocol/graph-node:latest"
>        ],
>        "RepoDigests": [
>           "graphprotocol/graph-node@sha256:787665b2e82c915e407f9eb14ea59c78d840d05bf186a20d6e32a8a4b0b7a8c5"
>        ],
Also, The version of Graphman is `0.32.0 (2023-08-21)`.

Before I make down the docker-compose, I can see the old blocks are syncing and I can see some logs like these:
> subgraph-docker-graph-node-1         | Oct 17 09:27:58.029 INFO Scanning blocks [6145095, 6145472], range_size: 378, sgd: 1, subgraph_id: QmeZJwXBbMzxn9EecUzZfYpYAVqd8v4YozCwdXXzc1pTEq, component: BlockStream

> subgraph-docker-graph-node-1         | Oct 16 15:35:59.671 INFO Done processing trigger, gas_used: 115773562, data_source: Contract, handler: handleApproval, total_ms: 1, transaction: 0x5b4d…425c, address: 0x1e4a…d41d, signature: Approval(indexed address,indexed address,uint256), sgd: 1, subgraph_id: QmeZJwXBbMzxn9EecUzZfYpYAVqd8v4YozCwdXXzc1pTEq, component: SubgraphInstanceManager

>subgraph-docker-graph-node-1         | Oct 16 15:35:59.672 INFO Done processing trigger, gas_used: 115587310, data_source: Contract, handler: handleTransfer, total_ms: 1, transaction: 0x5b4d…425c, address: 0x1e4a…d41d, signature: Transfer(indexed address,indexed address,uint256), sgd: 1, subgraph_id: QmeZJwXBbMzxn9EecUzZfYpYAVqd8v4YozCwdXXzc1pTEq, component: SubgraphInstanceManager

>subgraph-docker-graph-node-1         | Oct 16 15:35:59.672 INFO Applying 3 entity operation(s), block_hash: 0xc93887e94a726023f93e8f92721137556ceeb3248f49cd04aa07d6a0e3fb1607, block_number: 6322141, sgd: 1, subgraph_id: QmeZJwXBbMzxn9EecUzZfYpYAVqd8v4YozCwdXXzc1pTEq, component: SubgraphInstanceManager

Also, When I continuously watch the `graphmnan info -s` output I can see both `latest block` and `chain head blocks` are increasing over time.
But after I start the docker compose again I cannot see the above logs again. The only thing I can see are these logs:

> subgraph-docker-graph-node-1         | Oct 17 08:27:39.972 INFO Syncing 2 blocks from Ethereum, code: BlockIngestionStatus, blocks_needed: 2, blocks_behind: 2, latest_block_head: 6338690, current_block_head: 6338688, provider: polygon-zkevm2, component: EthereumPollingBlockIngestor

Also, When I continuously watch the `graphmnan info -s` output again, the `latest block` does not increase anymore. Just the `chain head blocks` field is increasing over time.

**Comment by incrypto32** [comment]
And there is no errors for that subgraph? Do you see a log like "Starting or restarting subgraph" after the docker restart?

**Comment by mehditvk8** [comment]
No. There is no error. I didn't see logs like that. @incrypto32

**Comment by 1997roylee** [comment]
The graph node stop working after restart the docker
